src/ui/public/template_vis_type/template_renderbot.js

```diff
@@ -11,6 +11,10 @@ export default function TemplateRenderbotFactory(Private, $compile, $rootScope)
     this.$scope.vis = vis;
     this.$scope.uiState = uiState;
 
+    if (this.vis.listeners.renderComplete) {
```

Comment:
I don't like these instance checks. We should give all visualizations the renderComplete API.